implementations/rust/ockam/ockam_command/src/orchestrator/show.rs

```diff
@@ -0,0 +1,104 @@
+use crate::util::api::CloudOpts;
+use crate::util::node_rpc;
+use crate::{fmt_info, CommandGlobalOpts};
+use clap::Args;
+use colorful::Colorful;
+use miette::{miette, IntoDiagnostic};
+use minicbor::{Decode, Decoder, Encode};
+use ockam::identity::{SecureChannelOptions, TrustIdentifierPolicy};
+use ockam::{route, Context, MessageSendReceiveOptions, Node, TcpConnectionOptions, TcpTransport};
+use ockam_api::cli_state::StateDirTrait;
+use ockam_api::nodes::NodeManager;
+use ockam_core::api::{Request, Response, Status};
+use std::time::Duration;
+
+/// Retrieve the version information from Orchestrator nodes
+#[derive(Clone, Debug, Args)]
+pub struct ShowCommand {
+    /// Override default timeout (in seconds)
+    #[arg(long, default_value = "10")]
+    pub timeout: u64,
+}
+
+impl ShowCommand {
+    pub fn run(self, options: CommandGlobalOpts) {
+        node_rpc(get_orchestrator_version, (options, self))
+    }
+}
+
+async fn get_orchestrator_version(
+    ctx: Context,
+    (opts, cmd): (CommandGlobalOpts, ShowCommand),
+) -> miette::Result<()> {
+    let controller_addr = CloudOpts::route();
+    let controller_tcp_addr = controller_addr.to_socket_addr().into_diagnostic()?;
+    let controller_identifier = NodeManager::load_controller_identifier().into_diagnostic()?;
+    let tcp = TcpTransport::create(&ctx).await.into_diagnostic()?;
+    let connection = tcp
+        .connect(controller_tcp_addr, TcpConnectionOptions::new())
+        .await
+        .into_diagnostic()?;
+    let node = {
+        let identities_vault = opts.state.vaults.default()?.get().await?;
+        let identities_repository = opts.state.identities.identities_repository().await?;
+        Node::builder()
+            .with_identities_vault(identities_vault)
+            .with_identities_repository(identities_repository)
+            .build(ctx)
+            .await
+            .into_diagnostic()?
+    };
+    let identifier = opts.state.identities.default()?.identifier();
+    let secure_channel_options = SecureChannelOptions::new()
+        .with_trust_policy(TrustIdentifierPolicy::new(controller_identifier))
+        .with_timeout(Duration::from_secs(50));
```

Comment:
Shouldn't the timeout argument be used here as well? The user interpretation of the `timeout` argument is probably going to be "how long do I give this command to execute before giving up?". If we reuse the timeout there we potentially double the max execution time but at least there is an arg to play with.

Reply:
Yeah, I found this fixed timeout weird. If we reuse it here, perhaps we should increment the default value of 10 seconds up to 30.